lager/debug/http_server.hpp

```diff
@@ -230,38 +468,78 @@ class http_debug_server
                       std::string resources_path)
         : argc_{argc}
         , argv_{argv}
+        , port_{port}
         , resources_path_{std::move(resources_path)}
-        , server_{httpserver::create_webserver(port)}
-    {}
+    {
+    }
 
     template <typename Debugger>
     handle<Debugger>& enable(Debugger)
     {
         assert(!handle_);
-        assert(!server_.is_running());
         using handle_t = handle<Debugger>;
         auto hdl_      = std::unique_ptr<handle_t>(
             new handle_t{argc_, argv_, resources_path_});
         auto& hdl = *hdl_;
-        server_.register_resource("/api/step/{cursor}", &hdl.step_resource_);
-        server_.register_resource("/api/goto/{cursor}", &hdl.goto_resource_);
-        server_.register_resource("/api/undo", &hdl.undo_resource_);
-        server_.register_resource("/api/redo", &hdl.redo_resource_);
-        server_.register_resource("/api/pause", &hdl.pause_resource_);
-        server_.register_resource("/api/resume", &hdl.resume_resource_);
-        server_.register_resource("/api/?", &hdl.root_resource_);
-        server_.register_resource("/?.*", &hdl.gui_resource_);
+
+        detail::http::router router;
+
+        namespace ph = std::placeholders;
+
+        router.register_resource(
+            beast::http::verb::get,
+            "/api/?",
+            std::bind(&handle_t::root_resource_, &hdl, ph::_1));
```

Comment:
I think part of this design, with the resources being methods of the handle, and the handle separate from the server, was motivated by the way the old http server library worked, and it could be made more simple or elegant now... Anyway this is not a requirement to change this, just food for thought.